oss-licenses-plugin/src/main/groovy/com/google/android/gms/oss/licenses/plugin/OssLicensesPlugin.groovy

```diff
@@ -16,75 +16,76 @@
 
 package com.google.android.gms.oss.licenses.plugin
 
+import com.android.build.api.artifact.SingleArtifact
 import com.android.build.gradle.api.BaseVariant
 import org.gradle.api.Plugin
 import org.gradle.api.Project
+import org.slf4j.LoggerFactory
 
 class OssLicensesPlugin implements Plugin<Project> {
+
+    private static final logger = LoggerFactory.getLogger(DependencyTask.class)
+
     void apply(Project project) {
-        def getDependencies = project.tasks.create("getDependencies",
-                DependencyTask)
-        def dependencyOutput = new File(project.buildDir,
-            "generated/third_party_licenses")
-        def generatedJson = new File(dependencyOutput, "dependencies.json")
-        getDependencies.setProject(project)
-        getDependencies.outputDir = dependencyOutput
-        getDependencies.outputFile = generatedJson
+        def variantTolicenseTaskMap = new HashMap<String, LicensesTask>()
+        project.androidComponents {
+            onVariants(selector().all(), { variant ->
+                def baseDir = new File(project.buildDir,
+                        "generated/third_party_licenses/${variant.name}")
+                def dependenciesJson = new File(baseDir, "dependencies.json")
 
-        def resourceOutput = new File(dependencyOutput, "/res")
-        def outputDir = new File(resourceOutput, "/raw")
-        def licensesFile = new File(outputDir, "third_party_licenses")
-        def licensesMetadataFile = new File(outputDir,
-                "third_party_license_metadata")
-        def licenseTask = project.tasks.create("generateLicenses", LicensesTask)
+                def dependencyTask = project.tasks.register(
+                        "${variant.name}OssDependencyTask",
+                        DependencyTask.class) {
+                    it.outputDir = baseDir
```

Comment:
see  comment above, probably not necessary.

Reply:
Done.